apps/client/app/services/auth.server.ts

```diff
@@ -0,0 +1,127 @@
+import bcrypt from "bcryptjs";
+import { json, redirect, TypedResponse } from "@remix-run/node";
+import {
+  getUserByAttribute,
+  storeUser,
+} from "~/models/user.server";
+import type { User } from "~/types/auth";
+import { JWT } from "~/utils/jwt";
+import { accessTokenCookie, createUserSession, getAccessToken, getRefreshToken, refreshTokenCookie } from "./session.server";
+
+export const login = (email: string, password: string): TypedResponse<{ accessToken?: string, refreshToken?: string, error?: string }> => {
+  const jwt = new JWT();
+  const user = getUserByAttribute("email", email);
+
+  if (user) {
+    if (bcrypt.compareSync(password, user.password)) {
+      return json({
+        accessToken: jwt.generateAccessToken(user.id),
+        refreshToken: jwt.generateRefreshToken(user.id),
+      });
+    } else {
+      return json({ error: "Invalid password" }, { status: 401 });
```

Comment:
minor thing, Its not a good practice to return a message related to `invalid password` for being generic due to brute force hack.

```typescript
  try {
    const jwt = getJwtInstance();
    const user = await getUserByEmail(email);

    if (user && bcrypt.compareSync(password, user.password)) {
      const accessToken = jwt.generateAccessToken(user.id);
      const refreshToken = jwt.generateRefreshToken(user.id);
      return json({ accessToken, refreshToken });
    }
    
    return json({ error: "Invalid email or password" }, { status: 401 });
  } catch (error) {
    return json({ error: `An error occurred during login: ${error.message}` }, { status: 500 });
  }
  ```